src/libraries/Common/tests/Tests/System/TimeProviderTests.cs

```diff
@@ -0,0 +1,380 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace Tests.System
+{
+    public class TimeProviderTests
+    {
+        [Fact]
+        public void TestUtcSystemTime()
+        {
+            DateTimeOffset dto1 = DateTimeOffset.UtcNow;
+            DateTimeOffset providerDto = TimeProvider.System.UtcNow;
+            DateTimeOffset dto2 = DateTimeOffset.UtcNow;
+
+            Assert.InRange(providerDto.Ticks, dto1.Ticks, dto2.Ticks);
+            Assert.Equal(TimeSpan.Zero, providerDto.Offset);
+        }
+
+        [Fact]
+        public void TestLocalSystemTime()
+        {
+            DateTimeOffset dto1 = DateTimeOffset.Now;
+            DateTimeOffset providerDto = TimeProvider.System.LocalNow;
+            DateTimeOffset dto2 = DateTimeOffset.Now;
```

Comment:
Super unlikely, but presumably this is subject to rare failures in CI, if this were to run at exactly the wrong moment with a daylight savings time fall back?

Reply:
We always seem to have some failures every daylight transition.

Reply:
Yes, I should fix that :blush: